app/models/repository.rb

```diff
@@ -16,11 +17,33 @@ class Repository < ApplicationRecord
   acts_as_taggable
   acts_as_taggable_on :tags_for_repository
 
+  after_initialize :create_repository_identicon
+  after_create :clean_files
+
   def self.search_repo(key_word, language)
```

Comment:
No need to write it like search_repo. If you are in Repository model and writing a search method then it is obviously for search in repo. So just use `search` .